### PR TITLE
ROC-6126: Citizen initiate the payment for claim

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link
[ROC-6126](url)

### Change description
 Citizen initiate the payment for claim.
Please use env variable:
```
export FEATURE_INVERSION_OF_CONTROL=true
```

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/claim/routes/initiate-payment.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      } else {
+        res.redirect(Paths.confirmationPage.evaluateUri({ externalId: externalId }))
+      }
+    } catch (err) {
+      if (err.statusCode === HttpStatus.NOT_FOUND) {
+        const nextUrl: string = await claimStoreClient.initiatePayment(draft, user)
+        res.redirect(nextUrl)
+      } else {
+        logger.error(`claim with external id ${externalId} not found, redirecting user to check and send`)
+        next(err)
```
Comment (andrewwa-kainos) [inline src/main/features/claim/routes/initiate-payment.ts]: Missing redirection?

File: src/test/app/claims/claimStoreClient.ts [hunk 2]
```diff
@@ -164,5 +168,73 @@ describe('ClaimStoreClient', () => {
...
+        mockInternalServerErrorOnInitiatePayment()
+        try {
+          await claimStoreClient.initiatePayment(claimDraft, claimant)
+        } catch (err) {
+          expect(err.statusCode).to.equal(HttpStatus.INTERNAL_SERVER_ERROR)
+          expect(err.error).to.equal('An unexpected error occurred')
+          return
+        }
+
+        expect.fail() // Exception should have been thrown due to 500 response code
```
Comment (andrewwa-kainos) [inline src/test/app/claims/claimStoreClient.ts]: I find this clearer inside the `try` block - emphasises that the previous code bombed out, as expected

File: src/test/features/claim/routes/check-and-send.ts [hunk 3]
```diff
@@ -178,31 +207,41 @@ describe('Claim issue: check and send page', () => {
 
       it('Should validate that a claim made by soleTrader against company and their details', async () => {
         draftStoreServiceMock.resolveFind('claim',
-          { claimant: { ...draftStoreServiceMock.sampleClaimDraftObj.claimant, partyDetails: claimantSoleTraderDetails }, defendant: { ...draftStoreServiceMock.sampleClaimDraftObj.defendant, partyDetails: companyDetails } })
+          {
+            claimant: {
+              ...draftStoreServiceMock.sampleClaimDraftObj.claimant,
+              partyDetails: claimantSoleTraderDetails
+            }, defendant: { ...draftStoreServiceMock.sampleClaimDraftObj.defendant, partyDetails: companyDetails }
```
Comment (andrewwa-kainos) [inline src/test/features/claim/routes/check-and-send.ts]: If we're formatting, let's format consistently!

File: src/main/features/claim/routes/initiate-payment.ts (hunk 1 above)
Comment (sabahirfan) [reply src/main/features/claim/routes/initiate-payment.ts]: good catch. done.

**Pull request merged**